src/cdk/a11y/focus-monitor.spec.ts

```diff
@@ -10,11 +10,15 @@ import {A11yModule} from './index';
 describe('FocusMonitor', () => {
   let fixture: ComponentFixture<PlainButton>;
   let buttonElement: HTMLElement;
-  let buttonRenderer: Renderer2;
   let focusMonitor: FocusMonitor;
   let changeHandler: (origin: FocusOrigin) => void;
 
   beforeEach(() => {
+    spyOn(window, 'addEventListener').and.callThrough();
+    spyOn(window, 'removeEventListener').and.callThrough();
+    spyOn(document, 'addEventListener').and.callThrough();
+    spyOn(document, 'removeEventListener').and.callThrough();
```

Comment:
This will permanently make these things spies for the rest of the test run; I'm not sure there's a good way to test global listener deregistration.

Reply:
I believe they are automatically cleaned up at the end of the describe block. There's some discussion about it here: https://github.com/jasmine/jasmine/issues/1281